core/native/src/main/scala-2/java/text/Normalizer.scala

```diff
@@ -0,0 +1,74 @@
+package java.text
+
+import java.nio.CharBuffer
+import java.nio.charset.Charset
+import scala.annotation.nowarn
+import scala.scalanative.libc.stdlib
+import scala.scalanative.unsafe._
+
+object Normalizer {
+
+  def normalize(src: CharSequence, form: Form): String =
+    if (src == null || form == null)
+      throw new NullPointerException
+    else
+      Zone { implicit z =>
+        import Form._
+        import utf8proc._
+
+        val cstr = form match {
+          case NFC  => utf8proc_NFC(toCString(src))
+          case NFD  => utf8proc_NFD(toCString(src))
+          case NFKC => utf8proc_NFKC(toCString(src))
+          case NFKD => utf8proc_NFKD(toCString(src))
+          case _    => throw new IllegalArgumentException // should never happen
+        }
+
+        val normalized = fromCString(cstr) // TODO can be further optimized
```

Comment:
utf8proc offers a lower-level [`utf8proc_map`](https://github.com/JuliaStrings/utf8proc/blob/2484e2ed5e1d9c19edcccf392a7d9920ad90dfaf/utf8proc.h#L691) method which directly returns the length of the normalized string. This would allow to skip an extra loop over the `cstr` to calculate the length based on the NULL-termination.